uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/toil-request-ctrl.js

```diff
@@ -0,0 +1,160 @@
+define([
+  'common/lodash',
+  'leave-absences/shared/modules/controllers',
+  'leave-absences/shared/controllers/request-ctrl',
+  'leave-absences/shared/models/instances/toil-leave-request-instance',
+], function (_, controllers) {
+  controllers.controller('ToilRequestCtrl', [
+    '$controller', '$log', '$q', '$uibModalInstance', 'api.optionGroup', 'AbsenceType', 'directiveOptions', 'TOILRequestInstance',
+    function ($controller, $log, $q, $modalInstance, OptionGroup, AbsenceType, directiveOptions, TOILRequestInstance) {
+      $log.debug('ToilRequestCtrl');
+
+      var parentRequestCtrl = $controller('RequestCtrl'),
+        vm = Object.create(parentRequestCtrl);
+
+      vm.directiveOptions = directiveOptions;
+      vm.$modalInstance = $modalInstance;
+      vm.leaveType = 'toil';
+      vm.initParams = {};
+
+      /**
+       * Calculate change in balance, it updates local balance variables.
+       * It overrides the parent's implementation
+       *
+       * @return {Promise} empty promise if all required params are not set otherwise promise from server
+       */
+      vm.calculateBalanceChange = function () {
+        if (vm.request.toil_to_accrue) {
+          vm.loading.calculateBalanceChange = true;
+          vm.balance.change.amount = -vm.request.toil_to_accrue;
+          vm.balance.closing = vm.balance.opening + vm.balance.change.amount;
+          vm.uiOptions.showBalance = true;
+          vm.loading.calculateBalanceChange = false;
+        }
+      };
+
+      /**
+       * Calculates toil expiry date.
+       * TODO It will be based on from date for both single and multiple days for now.
+       *
+       * @return {Promise}
+       */
+      vm.calculateToilExpiryDate = function () {
+        if (!vm.request.from_date) {
+          vm.error = 'Please select from date to find expiry date';
+          return $q.reject(vm.error);
+        }
+
+        return AbsenceType.calculateToilExpiryDate(vm.request.type_id, vm.request.from_date)
+          .then(function (expiryDate) {
+            vm.expiryDate = expiryDate;
+          });
+      };
+
+      /**
+       * Checks if submit button can be enabled for user and returns true if successful
+       *
+       * @return {Boolean}
+       */
+      vm.canSubmit = function () {
+        return !!vm.request.duration && !!vm.request.toil_to_accrue &&
```

Comment:
I talking about this function, here vm and this used in combination
` return !!vm.request.duration && !!vm.request.toil_to_accrue &&
          !!this.request.from_date && !!this.request.to_date;`